feat(curriculum): add case converter project to teach list comprehension (#52569)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-list-comprehension-by-building-a-case-converter-program/657e928716b77b2277980276.md

---
id: 657e928716b77b2277980276
title: Step 1
challengeType: 20
dashedName: step-1
---

# --description--

> In this project, you are going to learn about list comprehensions in Python by building a program that can take `camelCase` or `PascalCase` formatted string and convert that to a `snake_case` formatted string. 

Reviewer: ```suggestion
In this project, you are going to learn about list comprehensions in Python by building a program that can take a `camelCase` or `PascalCase` formatted string and convert that to a `snake_case` formatted string. 
```

> List comprehensions in Python are concise way to construct a list without the usage of loops or the `append()` method. Apart from being briefer, list comprehensions often run faster.

Reviewer: ```suggestion
List comprehensions in Python are a concise way to construct a list without using loops or the `.append()` method. Apart from being briefer, list comprehensions often run faster.
```

> Start defining a new function named `convert_to_snake_case` that accepts a string named `pascal_or_camel_cased_string` as input.

Reviewer: ```suggestion
Start defining a new function named `convert_to_snake_case()` that accepts a string named `pascal_or_camel_cased_string` as input.
```

# --hints--

> You should define a new function named `convert_to_snake_case`.

Reviewer: ```suggestion
You should define a new function named `convert_to_snake_case()`.
```

```js
({
  test: () => {
    assert(__pyodide.runPython(`
      import inspect
      a = __locals.get("convert_to_snake_case")
      inspect.isfunction(a)
    `));
  }
})
```

> Your function should accept a parameter named `pascal_or_camel_cased_string`.

Reviewer: ```suggestion
Your function should take a parameter named `pascal_or_camel_cased_string`.
```
